ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/QueryCompactor.java

```diff
@@ -272,5 +278,16 @@ static void removeFilesForMmTable(HiveConf conf, AcidDirectory dir) throws IOExc
         fs.delete(dead, true);
       }
     }
+
+    static void overrideConfProps(HiveConf conf, CompactionInfo ci, Map<String, String> properties) {
+      Stream.of(properties, ci.properties != null ? new StringableMap(ci.properties) : null)
```

Comment:
no, nee for null check

Reply:
There are 2 implementations of StringableMap. One has the null check within, whereas the other one does not have it. I thought there was only one implementation. 
https://github.com/apache/hive/blob/master/common/src/java/org/apache/hadoop/hive/common/StringableMap.java
https://github.com/apache/hive/blob/master/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/utils/StringableMap.java

Reply:
Removed the following file - https://github.com/apache/hive/blob/master/common/src/java/org/apache/hadoop/hive/common/StringableMap.java
 
However, regarding the behaviour of toString wherein null is given a length of -1. I have retained the same behaviour moving forward mainly because of [HIVE-17391](https://issues.apache.org/jira/browse/HIVE-17391).